src/brsTypes/components/RoSGNode.ts

```diff
@@ -48,22 +50,28 @@ class Field {
     }
 
     setValue(value: BrsType) {
-        if (this.alwaysNotify || this.value !== value) {
+        let oldValue =  this.value;
+        this.value = value;
+        if (this.alwaysNotify || oldValue !== value) {
             this.observers.map(this.executeCallbacks);
         }
-        this.value = value;
     }
 
     addObserver(interpreter: Interpreter, callable: Callable) {
         let brsCallback: BrsCallback = {
             interpreter: interpreter,
+            environment: interpreter.environment,
```

Comment:
This callback didn't have the correct environment, because the `environment` property in the `Interpreter` is mutable and changes often, so I opted to save the environment as well.